foundation-extensions/mobile-core/mobile-core-api-reference.md

```diff
@@ -1,1182 +1,1805 @@
+{% endhint %}
+
+{% tabs %}
+{% tab title="iOS (AEP 3.x)" %}
+
+**Swift**
+
+### setAppGroup
+
+**Syntax**
+
+```swift
+public static func setAppGroup(_ group: String?)
+```
+
+**Example**
+
+```swift
+MobileCore.setAppGroup("appGroupId")
+```
+
+**Objective-C**
+
+### setAppGroup
+
+**Syntax**
+```swift
+@objc(setAppGroup:)
+public static func setAppGroup(_ group: String?)
+```
+
+**Example**
+
+```objectivec
+[AEPMobileCore setAppGroup:@"app-group-id"];
+```
+
+{% endtab %}
+
+{% tab title="iOS (ACP 2.x)" %}
+**Objective-C**
+
+### setAppGroup
+
+**Syntax**
+
+```swift
+public static func setAppGroup(_ group: String?)
+```
+
+**Example**
+
+```text
+[ACPCore setAppGroup:@"app-group-id"];
+```
+
+**Swift**
+
+### setAppGroup
+
+**Syntax**
+
+```swift
+public static func setAppGroup(_ group: String?)
+```
+
+**Example**
+
+```swift
+ACPCore.setAppGroup("app-group-id")
+```
+{% endtab %}
+
+{% tab title="Xamarin" %}
+**C\#**
+
+### setAppGroup
+
+**Syntax**
+
+```csharp
+public static void SetAppGroup (string appGroup);
+```
+
+**Example**
+
+```csharp
+ACPCore.SetAppGroup("app_group");
+```
+{% endtab %}
+{% endtabs %}
+
+## setApplication \(Android only\)
+
+When building Android applications, the `android.app.Application` reference must be passed to Mobile SDK, which allows Mobile SDK to access the `android.app.Context` and monitor the lifecycle of the Android application.
+
+{% hint style="warning" %}
+Android applications must call `MobileCore.setApplication()` before calling any other Mobile SDK API.
+{% endhint %}
+
+### setApplication
+
+You can use the `setApplication` method to pass the Android `Application` instance to Mobile SDK.
+
+{% tabs %}
+{% tab title="Android" %}
+**Java**
+
+### setApplication
+
+**Syntax**
+
+```java
+public static void setApplication(final Application app)
+```
+
+**Example**
+
+```java
+public class CoreApp extends Application {
+
+   @Override
+   public void onCreate() {
+      super.onCreate();
+      MobileCore.setApplication(this);
+      MobileCore.start(null);
+   }
+}
+```
+{% endtab %}
+
+{% tab title="Xamarin" %}
+**C\#**
+
+### setApplication
+
+**Example**
+
+```csharp
+public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
+{
+  protected override void OnCreate(Bundle savedInstanceState)
+  {
+    base.OnCreate(savedInstanceState);
+    ACPCore.Application = this.Application;
+    ACPCore.Start(null);
+  }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+
+
+
+
+
+
+
+## setLogLevel
+
+The logging APIs allow log messages to be tagged and filtered with the Mobile SDK log messages. This allows application developers to filter the logged messages based on the current logging mode.
+
+Application developers can use the `setLogLevel` method to filter the log messages that are coming from the Mobile SDK.
+
+From least to most verbose, the order of Mobile SDK logging modes is as follows:
+
+* ERROR
+* WARNING
+* DEBUG
+* VERBOSE
```

Comment:
should we add VERBOSE / TRACE since we use trace in AEP?

Reply:
How about

```
* VERBOSE (ACP 2.x) / TRACE (AEP 3.x)
```

Reply:
This seems to be the top level description for all platforms not only iOS. Can we use `VERBOSE / TRACE` here and then detail the actual values under the corresponding tab?